src/shared/remoteSession.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Settings } from '../shared/settings'
+
+export async function openRemoteTerminal(
+    options: vscode.TerminalOptions,
+    onClose: () => void,
+    onError: (err: unknown) => void
```

Comment:
Let the error bubble up and use `.catch`?

Reply:
Is the main advantage of using `.catch` here to reduce the # of parameters, and therefore making the function easier to understand and simpler?

Reply:
Both. It's simpler and now your error handler can be async if needed.